currency/pair.go

```diff
@@ -150,11 +140,22 @@ func MatchPairsWithNoDelimiter(currencyPair string, pairs Pairs, pairFmt PairFor
 
 // GetFormatting returns the formatting style of a pair
 func (p Pair) GetFormatting() (PairFormat, error) {
-	if p.Base.UpperCase != p.Quote.UpperCase {
-		return PairFormat{}, fmt.Errorf("%w casing mismatch", errPairFormattingInconsistent)
+	if p.Base.isCaseSensitive() && p.Quote.isCaseSensitive() && (p.Base.upperCase != p.Quote.upperCase) {
+		return EMPTYFORMAT, fmt.Errorf("%w casing mismatch", errPairFormattingInconsistent)
+	}
+	upper := p.Base.isCaseSensitive() && p.Base.upperCase || p.Quote.isCaseSensitive() && p.Quote.upperCase
```

Comment:
I think you don't need to check isCaseSensitive() like this.
In each of the `a && b` tests `b` can only be true if `a` is.
In which case, you only need to test `b`
```suggestion
	upper := p.Base.upperCase || p.Quote.upperCase
```

Reply:
What a guy! 🔥